src/Common/src/TypeSystem/Interop/IL/Marshaller.cs

```diff
@@ -351,6 +350,7 @@ protected TypeDesc GetDefaultNativeType(NativeType nativeType)
 
                 case MarshallerKind.UnicodeString:
                     return GetDefaultNativeType(NativeType.SysInt); // We pin UniCodeString, so native type will be SysInt
```

Comment:
Some places use SysInt for unmanaged pointers, other places (like the one your have fixed above) are using the actual pointer types for unmanaged pointers. Could you please fix it to be consistent?